### PR TITLE
Remove optionality from added_by_address

### PR DESCRIPTION
**Pull request by zombieobject**
It was discussed amongst the team and agreed that the `added_by_address` String value on Groups should not be optional.

### CODE REVIEW

File: xmtp_mls/src/subscriptions.rs [hunk 1]
```diff
@@ -134,9 +134,14 @@ where
...
                             )?;
                             // TODO update cursor
-                            MlsGroup::new(self, group_id, stream_info.convo_created_at_ns, None)
-                                .process_stream_entry(envelope)
-                                .await
+                            MlsGroup::new(
+                                self,
+                                group_id,
+                                stream_info.convo_created_at_ns,
+                                "placeholder_address".to_string(),
```
[the remark this answers is not in the document]
Comment (richardhuaaa) [reply xmtp_mls/src/subscriptions.rs:141]: I don't think you should use a placeholder here, the surrounding code often gives a clue as to how to do things. Two ways to address this:

1. Just like `convo_created_at_ns`, add it to `group_id_to_info`.
2. (Recommended, requires a small refactor) - don't store `added_by` on the `MlsGroup`, that way you can remove the property from the constructor. Query the DB for `added_by` whenever it is needed

**Review by nplasterer — APPROVED** [review]
This looks much better! Kind of what I had in mind initially. Glad we could get there.

File: xmtp_mls/src/storage/encrypted_store/group.rs [hunk 2]
```diff
@@ -116,6 +117,14 @@ impl DbConnection<'_> {
         Ok(self.raw_query(|conn| query.load(conn))?)
     }
 
+    /// Return a single group that matches the given ID
+    pub fn find_group(&self, id: Vec<u8>) -> Result<Vec<StoredGroup>, StorageError> {
+        let mut query = dsl::groups.order(dsl::created_at_ns.asc()).into_boxed();
+        query = query.filter(dsl::id.eq(id));
+
+        Ok(self.raw_query(|conn| query.load(conn))?)
+    }
```
[the remark this answers is not in the document]
Comment (richardhuaaa) [reply xmtp_mls/src/storage/encrypted_store/group.rs:126]: Yes, I'd also recommend adding a test that validates an error is thrown if there isn't exactly one (not zero, not two) rows

**Review by richardhuaaa — APPROVED** [review]
Just Scott's comment left to address, but this looks great! Thanks for iterating with us
